extensions/ql-vscode/src/cli.ts

```diff
@@ -576,6 +582,46 @@ export class CodeQLCliServer implements Disposable {
     }
   }
 
+  static async parseSarif(interpretedResultsPath: string) : Promise<sarif.Log> {
```

Comment:
This function doesn't require the CodeQL CLI. I think a better place for it would be `sarif-utils.ts`. Then you can place your tests for it in `sarif-utils.test.ts` and make them "pure" unit tests, i.e. completely independent of the CodeQL CLI and VS Code API. This is nice because it shows that you're testing the JSON parsing behaviour in isolation, regardless of how it integrates into CodeQL commands and the rest of the extension.

Reply:
Yes. That makes more sense. It was originally my suggestion to make this function static.